libraries/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts

```diff
@@ -541,6 +545,17 @@ export class PhasedScriptAction extends BaseScriptAction<IPhasedCommandConfig> {
         countNoOp: 0
       };
 
+      let buildTimings: IBuildTimeRecord[] = [];
```

Comment:
Could you move all the logic in the `if (this.parser.telemetry)` to be the implementation of a single `TelemetryPlugin` or similar that also includes all the logic for recording the graph and the build times?